src/Extension/Manager.php

```diff
@@ -124,9 +124,13 @@ public function add(ExtensionInterface $extension, DirectoryInterface $baseDir =
         // Set paths in the extension
         if ($baseDir !== null) {
             $extension->setBaseDirectory($baseDir);
+        } else {
+            $extension->setBaseDirectory($this->extFs->getDir($extension->getId()));
```

Comment:
I think it would make more sense to flip the condition here and skip the else block. 
```php
if ($baseDir === null) {
    $baseDir = $this->extFs->getDir($extension->getId());
}
$extension->setBaseDirectory($baseDir);
```